### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by johnhooks** [wp-blocks/wp-blocks#31] [Ideas]
@erikyo at the moment the monorepo is not set up for testing. I would prefer to use `vitest` for unit testing and `playwright` for end to end and component testing. I have some experience with `vitest` though I am totally new to `playwright`. It will be a learning process but I would rather work on getting that set up than figure out the `jest` config. In my experience `jest` in a monorepo is difficult to setup, `vitest` works with them very smoothly.

What do you think? We should implement some of the older style testing in the vinyl directory because I would like to start testing that before we probably have this system figured out. But that doesn't add a lot of overhead.

**Comment by erikyo** [comment]
I am very interested in learning how to use `playwright` instead because I was at a meetup where they presented it and I was pleasantly impressed. Never used `vitest` but I would gladly try, I have always used `jest` for the vast majority of times!

Reassuming, although I would have a certain preference for Playwright, in any case I would adapt, as long test does the job, it's fine by me!

**Comment by johnhooks** [comment]
Vitest in my experience has been so much easier to setup than jest, it just works without creating complicated configs
